plugins/destination/duckdb/client/read.go

```diff
@@ -0,0 +1,106 @@
+package client
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strings"
+
+	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/marcboeker/go-duckdb"
+)
+
+const (
+	readSQL = `SELECT %s FROM "%s" WHERE _cq_source_name = $1 order by _cq_sync_time asc`
+)
+
+func (*Client) createResultsArray(table *schema.Table) []any {
+	results := make([]any, 0, len(table.Columns))
+	for _, col := range table.Columns {
+		switch col.Type {
+		case schema.TypeBool:
+			var r *bool
```

Comment:
1. group the options with the same `r` together
2. may be a bit simplified by `append(results, new(*bool))` code

Reply:
I don't think it really matters, and it's similar to most other destination plugins, so will keep it this way